watchdog.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+/**
+ * This is a watch dog timer.  Think of it as a ticking
+ * timebomb which will throw an explosion when it hasn't
+ * been stopped or touched in `maxTime` seconds.  The
+ * `WatchDog` will throw an `Error` with `msg` if the
+ * timer is allowed to expire
+ */
+function WatchDog(maxTime, action) {
+  this.maxTime = maxTime;
+  if (typeof action === 'function') {
+    this.action = action;
+  } else {
+    this.action = function() {
+      throw new Error(action);
```

Comment:
It might be preferable to emit and `error` event on the object instead... if not cached it'll cause node to crash...